src/neo4j_genai/retrievers/filters.py

```diff
@@ -0,0 +1,345 @@
+                    f"Unsupported type: {type(val)} for value: {val}"
+                )
+        return value
+
+
+class NinOperator(InOperator):
+    CYPHER_OPERATOR = "NOT IN"
+
+
+class LikeOperator(Operator):
+    CYPHER_OPERATOR = "CONTAINS"
+
+    def cleaned_value(self, value):
+        if not isinstance(value, str):
+            raise ValueError(f"Expected string value, got {type(value)}: {value}")
+        return value.rstrip("%")
+
+
+class ILikeOperator(LikeOperator):
+    def lhs(self, field):
+        return f"toLower({self.node_alias}.`{field}`)"
+
+    def cleaned_value(self, value):
+        value = super().cleaned_value(value)
+        return value.lower()
+
+
+OPERATOR_PREFIX = "$"
+
+OPERATOR_EQ = "$eq"
+OPERATOR_NE = "$ne"
+OPERATOR_LT = "$lt"
+OPERATOR_LTE = "$lte"
+OPERATOR_GT = "$gt"
+OPERATOR_GTE = "$gte"
+OPERATOR_BETWEEN = "$between"
+OPERATOR_IN = "$in"
+OPERATOR_NIN = "$nin"
+OPERATOR_LIKE = "$like"
+OPERATOR_ILIKE = "$ilike"
+
+OPERATOR_AND = "$and"
+OPERATOR_OR = "$or"
+
+COMPARISONS_TO_NATIVE = {
+    OPERATOR_EQ: EqOperator,
+    OPERATOR_NE: NeqOperator,
+    OPERATOR_LT: LtOperator,
+    OPERATOR_LTE: LteOperator,
+    OPERATOR_GT: GtOperator,
+    OPERATOR_GTE: GteOperator,
+    OPERATOR_IN: InOperator,
+    OPERATOR_NIN: NinOperator,
+    OPERATOR_LIKE: LikeOperator,
+    OPERATOR_ILIKE: ILikeOperator,
+}
+
+
+LOGICAL_OPERATORS = {OPERATOR_AND, OPERATOR_OR}
+
+SUPPORTED_OPERATORS = (
+    set(COMPARISONS_TO_NATIVE).union(LOGICAL_OPERATORS).union({OPERATOR_BETWEEN})
+)
+
+
+class ParameterStore:
+    """
+    Store parameters for a given query.
+    Determine the parameter name depending on a parameter counter
+    """
+
+    def __init__(self):
+        self._counter = Counter()
+        self.params = {}
+
+    def _get_params_name(self, key="param"):
+        """Find parameter name so that param names are unique.
+        This function adds a suffix to the key corresponding to the number
+        of times the key have been used in the query.
+        E.g.
+        node.age >= $param_0 AND node.age <= $param_1
+
+        :param p:
+        :param counter:
+        :return:
+        """
+        # key = slugify(key.replace(".", "_"), separator="_")
+        param_name = f"{key}_{self._counter[key]}"
+        self._counter[key] += 1
+        return param_name
+
+    def add(self, key, value):
+        """This function adds a new parameter to the param dict.
+        It returns the name of the parameter to be used as a placeholder
+        in the cypher query, e.g. $param_0"""
+        param_name = self._get_params_name()
+        self.params[param_name] = value
+        return param_name
+
+
+def _single_condition_cypher(
+    field: str,
+    native_operator_class: Type[Operator],
+    value: Any,
+    param_store: ParameterStore,
+    node_alias: str,
+) -> str:
+    """Return Cypher for field operator value.
+
+    Args:
+        field: the name of the field being filtered
```

Comment:
```suggestion
        field: The name of the field being filtered
```

Don't forget to capitalize the first letter of the descriptions